### PR TITLE
move to uproxy-lib 18, with loggingprovider

### PR DESCRIPTION
**Pull request by trevj**
OK, so, yeah. This updates the repo to use uproxy-lib 18.

This has the new loggingprovider stuff so it's mostly a matter of updating the sample apps to add loggingprovider as a dependency and updating calls to `setConsoleFilter`.

@willscott: Does this look about right to you? Do you spot any unnecessary stuff in here? It's quite a big diff, would appreciate any suggestions you might have on trimming it.

@iislucas: I know you are working on some improvements to mocking with the move to require but hopefully this is not unacceptable.

[Review on Reviewable](https://reviewable.io/reviews/uproxy/uproxy-networking/188)

### CODE REVIEW

File: src/churn/mocks.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
 // Create a mock instance of Freedom.
 // We do this in a non-TypeScript file because the ambient module declaration
 // prevents us creating any variable called freedom in TypeScript-land.
-var freedom = jasmine.createSpyObj('freedom', ['core.log']);
-freedom.churn = jasmine.createSpy().and.returnValue(
-    jasmine.createSpyObj('churn', ['providePromises']));
-freedom['core.console'] = function() {
-  return window.console;
+var freedom = {
```
Comment (willscott) [inline src/churn/mocks.js:4]: we also have freedom mocks in uproxy-lib. is there a way to have them centralized in one place?

File: src/echo/freedom-module.ts [hunk 2]
```diff
@@ -2,6 +2,8 @@
 /// <reference path='../logging/logging.d.ts' />
 /// <reference path='../freedom/typings/freedom.d.ts' />
 
+freedom['loggingprovider']().setConsoleFilter(['*:D']);
```
Comment (willscott) [inline src/echo/freedom-module.ts:5]: this is default behavior per https://github.com/uProxy/uproxy-lib/blob/dev/src/loggingprovider/loggingprovider.ts#L33  Why needed?

**Comment by willscott** [comment]
Looks generally good to me - only real thought on trimming would be trying to centralize the mock to one shared location, and maybe removing the 'loggingprovider' interface from the demos that don't set a meaningful console filter.

File: src/churn/mocks.js (hunk 1 above)
Comment (trevj) [inline src/churn/mocks.js:4]: There is, and @iislucas is working on this as part of his big require change. So, we can benefit from that when the time comes to update this repo.

File: src/echo/freedom-module.ts (hunk 2 above)
Comment (trevj) [inline src/echo/freedom-module.ts:5]: It's often useful to adjust the logging level when working on new features or debugging an issue, so this is an example of how to do it.

**Comment by trevj** [comment]
Thanks for looking!

@iislucas I'll merge this if you don't have any objections after taking a quick look over?

**Comment by bemasc** [comment]
:+1:

I don't really know what's going on here, but if it works, it's fine by me.

**Pull request merged**